### ISSUE TITLE
Adding support for CMake

### ISSUE THREAD

**Issue by compor** [grisumbras/enum-flags#12]
I'm just exploring the possibility of adding support for CMake (while keeping support for Jam), 
in order to export the library for use in other CMake projects (something similar to [this](https://github.com/nlohmann/json) project or maybe more like [this](https://github.com/bombela/backward-cpp) one):

- as a standalone external dependency or 
- as a git submodule

At the moment I have a preliminary basic CMake file, but I could expand it and include running of unit tests as well.

Would a PR like that be considered?

**Comment by grisumbras** [comment]
If you are still interested, PR #18 adds installation of CMake config file, which allows using the project via `find_package` (see [here](../tree/master/test/cmake/CMakeLists.txt) for an example).

**Comment by compor** [comment]
Hey, sorry for the long delay in replying. Although I don't actively use this project anymore, I'll have a look at this PR soonish.

**Comment by barcharcraz** [comment]
The cmake exports are non-relocatable.

**Comment by grisumbras** [comment]
Indeed. I believe, I have the fix in the works, but first I need to come up with a good way to test for exports being relocatable.
